pkg/expr/commands_test.go

```diff
@@ -100,42 +101,83 @@ func Test_UnmarshalReduceCommand_Settings(t *testing.T) {
 
 func TestReduceExecute(t *testing.T) {
 	varToReduce := util.GenerateShortUID()
-	cmd, err := NewReduceCommand(util.GenerateShortUID(), randomReduceFunc(), varToReduce, nil)
-	require.NoError(t, err)
 
-	t.Run("should noop if Number", func(t *testing.T) {
+	t.Run("when mapper is nil", func(t *testing.T) {
+		cmd, err := NewReduceCommand(util.GenerateShortUID(), randomReduceFunc(), varToReduce, nil)
+		require.NoError(t, err)
+
+		t.Run("should noop if Number", func(t *testing.T) {
+			var numbers mathexp.Values = []mathexp.Value{
+				mathexp.GenerateNumber(util.Pointer(rand.Float64())),
+				mathexp.GenerateNumber(util.Pointer(rand.Float64())),
+				mathexp.GenerateNumber(util.Pointer(rand.Float64())),
+			}
+
+			vars := map[string]mathexp.Results{
+				varToReduce: {
+					Values: numbers,
+				},
+			}
+
+			execute, err := cmd.Execute(context.Background(), time.Now(), vars, tracing.NewFakeTracer())
+			require.NoError(t, err)
+
+			require.Len(t, execute.Values, len(numbers))
+			for i, value := range execute.Values {
+				expected := numbers[i]
+				require.Equal(t, expected.Type(), value.Type())
+				require.Equal(t, expected.GetLabels(), value.GetLabels())
+
+				expectedValue := expected.Value().(*mathexp.Number).GetFloat64Value()
+				actualValue := value.Value().(*mathexp.Number).GetFloat64Value()
+				require.Equal(t, expectedValue, actualValue)
+			}
+
+			t.Run("should add warn notices to every frame", func(t *testing.T) {
```

Comment:
We should see if works with only the first frame, and if it does only put it there. Else were could be repeating this string a lot if there are many items in the response.

Reply:
I can address it in a follow-up PR.